src/test.rs

```diff
@@ -107,3 +107,9 @@ fn positive_test_run_pass() {
 
     compiletest::run_tests(&config);
 }
+
+#[test]
+#[should_panic]
+fn panic() {
+    join(|| {}, || panic!("should panic"));
```

Comment:
I am vaguely concerned that having a panic in a test like this will infect the global schedule and affect other tests -- we may want to move these kinds of panic-y tests to run-fail tests using the `compiletest` infrastructure. We'd have to kind of copy-and-paste the `positive_tests_run_pass` test right above and change the `mode` variable to `fun-fail` and the directory to `neg-tests-run`. Then you can make `.rs` files that are expected to panic. I think they have to have a comment like `// error-pattern: foo`, where `foo` is a string we expected to see in the output.